tmt/base.py

```diff
@@ -1307,27 +1311,50 @@ def lint_manual_valid_markdown(self) -> LinterReturn:
     def lint_require_type_field(self) -> LinterReturn:
         """ T009: require fields should have type field """
 
-        filename = self.node.sources[-1]
-        metadata = tmt.utils.yaml_to_dict(self.read(filename))
+        self.node.get('require')
         missing_type = []
 
         # Check require items have type field
-        for require in metadata.get('require', []):
-            if isinstance(require, dict) and not require.get('type'):
-                missing_type.append(require)
+        complex_dependencies = [
+            dependency for dependency in self.node.get('require', [])
+            if isinstance(dependency, dict)
+            ]
+
+        if not complex_dependencies:
+            yield LinterOutcome.SKIP, 'library/file requirements not used'
+            return
+
+        missing_type = [
+            dependency for dependency in complex_dependencies if not dependency.get('type')
+            ]
 
-        if missing_type and not self.opt('fix'):
-            yield LinterOutcome.FAIL, 'requirements should specify type, library or file'
+        if not missing_type:
+            yield LinterOutcome.PASS, 'all library/file requirements specify type'
             return
 
-        if missing_type:
-            for require in missing_type:
-                require['type'] = 'file' if require.get('pattern') else 'library'
-            self.write(filename, tmt.utils.dict_to_yaml(metadata))
-            yield LinterOutcome.FIXED, 'added type to requirements'
+        if not self.opt('fix'):
+            yield LinterOutcome.FAIL, 'library/file requirements should specify type'
             return
 
-        yield LinterOutcome.PASS, 'all requirements have type field'
+        filename = self.node.sources[-1]
+        metadata = tmt.utils.yaml_to_dict(self.read(filename))
+
+        if not tmt.utils.is_key_origin(self.node, 'require') \
+                and all(dependency in metadata.get('require', []) for dependency in missing_type):
+            yield LinterOutcome.FAIL, \
+                'some library/file requirement are missing type, but inherited from test parent,' \
+                ' please, fix manually'
+            return
+
+        for dependency in metadata.get('require', []):
+            if not isinstance(dependency, dict) or dependency.get('type'):
+                continue
+
+            dependency['type'] = 'file' if dependency.get('pattern') else 'library'
+
+        # self.write(filename, tmt.utils.dict_to_yaml(metadata))
```

Comment:
So the `--fix` feature is disabled?

Reply:
Fixed in 1ebdbdcf, this looks like a left-over from refactoring.